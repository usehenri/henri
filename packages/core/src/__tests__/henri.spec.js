const Henri = require('../henri');

describe('henri', () => {
  describe('general', () => {
    beforeEach(() => {
      this.henri = new Henri({ runlevel: 1 });
    });

    it('should match snapshot', () => {
      delete this.henri.validator;
      this.henri.validator = {};

      expect(this.henri).toMatchSnapshot();
    });

    it('should have a reload method', () => {
      expect(this.henri.reload).toBeDefined();

      expect(this.henri.reload()).toBeTruthy();
    });

    it('should have a stop method', () => {
      expect(this.henri.stop).toBeDefined();

      expect(this.henri.stop()).toBeTruthy();
    });

    it('should register a global', () => {
      process.env.NODE_ENV = 'dev';

      const local = new Henri();

      expect(henri).toBeDefined();
      expect(local).toBeDefined();

      process.env.NODE_ENV = 'test';
    });

    it('should have a validator', () => {
      expect(this.henri.validator.isEmail('code@usehenri.io')).toBeTruthy();
    });
  });

  describe('with options', () => {
    it('should have runlevel 6 by default', () => {
      const henri = new Henri();

      expect(henri.runlevel).toEqual(6);
    });

    it('should accept different runlevel', () => {
      const henri = new Henri({ runlevel: 2 });

      expect(henri.runlevel).toEqual(2);
    });

    it('should have . as default folder', () => {
      process.env.NODE_ENV = 'dev';
      const henri = new Henri();

      expect(henri.prefix).toEqual('.');

      process.env.NODE_ENV = 'test';
    });

    it('should accept a new folder', () => {
      process.env.NODE_ENV = 'dev';
      const henri = new Henri({ cwd: './packages' });

      expect(henri.prefix).toEqual('./packages');
      process.env.NODE_ENV = 'test';
    });
  });

  describe('main module', () => {
    test('should not be null', () => {
      // eslint-disable-next-line global-require
      const start = require('../index');

      expect(typeof start).toEqual('function');
    });
  });
});
