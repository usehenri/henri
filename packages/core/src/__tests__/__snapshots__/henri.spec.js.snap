// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`henri general should match snapshot 1`] = `
Henri {
  "_middlewares": Array [],
  "consoleOnly": false,
  "cwd": [Function],
  "env": "test",
  "isDev": false,
  "isProduction": false,
  "isTest": true,
  "isTesting": false,
  "modules": Modules {
    "add": [Function],
    "henri": [Circular],
    "init": [Function],
    "initialized": false,
    "modules": Map {},
    "order": Array [
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
    ],
    "reload": [Function],
    "reloadable": Array [
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
    ],
    "stopOrder": Array [],
    "store": Array [
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
      Array [],
    ],
  },
  "pen": Pen {
    "_time": "42424242",
    "_timeSkipped": 0,
    "buffer": Array [],
    "consoleOnly": false,
    "henri": null,
    "inTesting": true,
    "info": [Function],
    "init": [Function],
    "initialized": false,
    "key": null,
    "longest": 12,
    "name": "unnamed",
    "notTest": false,
    "reloadable": false,
    "runlevel": 6,
    "setup": [Function],
    "start": [Function],
  },
  "prefix": "./packages/demo",
  "release": "0.42.0",
  "runlevel": 1,
  "settings": Object {
    "arch": "x64",
    "package": "0.42.0",
    "platform": "linux",
  },
  "status": Map {},
  "utils": Object {
    "bounce": Object {
      "background": [Function],
      "ignore": [Function],
      "isBoom": [Function],
      "isError": [Function],
      "isSystem": [Function],
      "rethrow": [Function],
    },
    "checkPackages": [Function],
    "clearConsole": [Function],
    "getColor": [Function],
    "stack": [Function],
    "syntax": [Function],
    "yarnExists": [Function],
  },
  "validator": Object {},
}
`;
